### ISSUE TITLE
Quotes automatically added to identifiers, causing problems

### ISSUE THREAD

**Issue by thebecwar** [duckdb/postgres_scanner#262]
### What happens?

Somewhere in the query parsing/preparation path, identifiers are being quoted. This is effectively making these identifiers case-sensitive, but the person writing the query has no control over it. As a user, I would expect that the queries sent to postgres use the same quoting as the query I input, but it doesn't seem to work like that.

Additionally the identifiers are inconsistently quoted. In a create table query, the schema and table name are quoted, but column names are not.  (see below) This causes problems with inserts since the schema cache seems to take the field names from the insert.

Happy to provide more information if needed. If this behavior is working as intended, is there any way to shut off quoting identifiers? If not, can the rules for what gets documented, and when the cache is updated be added to the docs? I'm happy to contribute the documentation if someone can point me to the code that's running the logic.

### To Reproduce

Using `SET pg_debug_show_queries=true;` to show the actual queries going to postgres. Comments in line with some commentary.

```
ATTACH '[connection string]' AS compute (TYPE POSTGRES);
CREATE SCHEMA IF NOT EXISTS helpers;

CREATE TABLE compute.helpers.test (field_Name INTEGER);
-- Actual query: CREATE TABLE "helpers"."test"(field_Name INTEGER); 
-- Here as expected the column name is not quoted, but I would expect that the schema and table are not quoted.

SELECT * FROM compute.helpers.test;
-- Actual query: SELECT "field_Name" FROM "helpers"."test";
-- This results in an error, since the original create table query did not quote the field name, but the select query does.

CALL pg_clear_cache();
SELECT * FROM compute.helpers.test;
-- Actual query: SELECT "field_name" FROM "helpers"."test"
-- This works, but took me a while to discover the behavior. When writing "SELECT *" I don't expect that I 
-- will have problems with column names.

-- An insert of the form:
INSERT INTO compute.helpers.test SELECT * FROM dataframe;
-- Where the dataframe has upper cased column names, causes the cache to think that the columns are upper-cased.
-- Of course this shouldn't matter, but the generated PG query quotes all the identifiers.
```

### OS:

Linux

### PostgreSQL Version:

16

### DuckDB Version:

1.0.0

### DuckDB Client:

Python

### Full Name:

Mark Becwar

### Affiliation:

Conduce Health

### Have you tried this on the latest `main` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by thebecwar** [comment]
Feel bad submitting a bug report here for something that may be WAI, so I want to add that the extension is generally awesome, and is allowing us to migrate from an expensive cloud data warehouse to a self-hosted postgres cluster.

**Comment by Mytherin** [comment]
Thanks for the report!

I think this is the same issue as https://github.com/duckdb/postgres_scanner/issues/250 - could you verify that the issue still occurs on the latest version (DuckDB v1.1.0)?

**Comment by thebecwar** [comment]
Thanks for the quick response. It looks like all of the schema cache issues are fixed by #250, but the identifiers are still being quoted which does lead to some unintuitive behavior - running a create table from duckdb where an identifier has an upper-case letter, makes the identifier case-sensitive, even where that query, run directly against postgres would not have made the identifier case sensitive.

```
CREATE TABLE compute.helpers.test (field_Name INTEGER);
-- Query to postgres: CREATE TABLE "helpers"."test"("field_Name" INTEGER);
-- Quotes being added here means that the field is case sensitive

INSERT INTO compute.helpers.test (FIELD_NAME) VALUES (1);
-- Query to postgres: COPY "helpers"."test" ("field_Name") FROM STDIN (FORMAT BINARY)
-- This works, since the field name is correctly mapped to its case-sensitive equivalent

SELECT * FROM compute.helpers.test;
-- Query to postgres: COPY (SELECT "field_Name" FROM "helpers"."test" ) TO STDOUT (FORMAT "binary");
```

The problem comes in when we want to directly use postgres to interact with the table created by duckdb. Since all of the identifiers in the create table query are quoted, and contain upper-case characters, they are case-sensitive.

```
% psql -d compute -c "SELECT field_name from helpers.test"
ERROR:  column "field_name" does not exist
LINE 1: SELECT field_name from helpers.test
               ^
HINT:  Perhaps you meant to reference the column "test.field_Name".
```

If everything is lowercase when running the create table, then we don't have that problem:
```
-- From duckdb:
CREATE TABLE compute.helpers.test (field_name INTEGER);
INSERT INTO compute.helpers.test (FIELD_NAME) VALUES (1);

Both of these queries now work:
% psql -d compute -c "SELECT field_Name from helpers.test;"
% psql -d compute -c "SELECT FIELD_NAME from HELPERS.TEST;"
```

As a workaround, I can create these schemas/tables with all lowercased identifiers, which seems to work. Up to you, but a note in the docs may be helpful. Something like:

> Note: When executing `CREATE` statements against the postgres database, identifiers have quotation marks added. If an identifier has upper-case characters, then the quotation marks will cause the identifier to be case sensitive. 
> `CREATE TABLE db.schema_name.table (Column_Name INTEGER);` will create a case-sensitive column, whereas `CREATE TABLE db.schema_name.table (column_name INTEGER);` will not. In either case, the postgres extension will correctly treat the column as case-insensitive by mapping the column name using the schema cache. For other use cases use the `postgres_execute` function, which does not modify the SQL being sent to postgres.

**Comment by Mytherin** [comment]
That's an unfortunate consequence of Postgres' confusing behavior around case sensitivity. To explain this better - Postgres does not have a concept of "case sensitive columns" - despite pretending to have this in the documentation. Instead, the way Postgres works is that it **lowercases** all unquoted columns. That is to say, this is what *actually* happens:

```sql
-- creates a table called "mytable" with a column called "mycolumn"
CREATE TABLE MyTable(MyColumn INT);
-- this now fails, because the table is not actually called "MyTable", but rather "mytable"
SELECT * FROM "MyTable";
-- ERROR:  relation "MyTable" does not exist
-- LINE 1: SELECT * FROM "MyTable";
--                       ^

``` 


Because of this behavior - there is no good solution here. Either we (1) maintain case by adding quotes, making Postgres itself require quotes when using these identifiers, or (2) lose all case and lower-case everything, effectively losing information that the user provided and losing the capability of round-tripping identifiers.

DuckDB actually inherited this case-sensitivity handling from Postgres, although we switched away from it early on since it was very confusing to users and caused a lot of problems when interacting with other tools. You can toggle this mode in DuckDB using `SET preserve_identifier_case=false`. This will make the system behave similarly to Postgres which will mimic this behavior also when interacting with Postgres.
